File: backend/app/application/use_case/order/cancel_order_usecase.py
from app.domain.repositories.order_repository import OrderRepository


class CancelOrderUseCase:
    def __init__(self, order_repo: OrderRepository):
        self.order_repo = order_repo

    def execute(self, order_id: int, user_id: int) -> dict:
        self.order_repo.cancel_order(order_id, user_id)
        return {"message": "Hủy đơn hàng thành công"}

File: frontend/user/lib/domain/usecase/order/cancel_order_usecase.dart
import 'package:b2205946_duonghuuluan_luanvan/core/error/failure.dart';
import 'package:dartz/dartz.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/repository/order/order_repository.dart';

class CancelOrderUseCase {
  final OrderRepository _repo;
  CancelOrderUseCase(this._repo);
  Future<Either<Failure, Unit>> call(int orderId) => _repo.cancelOrder(orderId);
}
